src/System.CommandLine/Binding/ModelBinder.cs

```diff
@@ -37,208 +26,210 @@ internal ModelBinder(IValueDescriptor valueDescriptor)
         internal Dictionary<IValueDescriptor, IValueSource> MemberBindingSources { get; } =
             new Dictionary<IValueDescriptor, IValueSource>();
 
-        protected ConstructorDescriptor FindModelConstructorDescriptor(
-            ConstructorInfo constructorInfo)
-        {
-            var cmpCtorDesc = new ConstructorDescriptor(constructorInfo,
-                // Parent does not matter for comparison and can be invalid.
-                parent: ModelDescriptor);
-            var cmpParamDescs = cmpCtorDesc.ParameterDescriptors
-                .Select(GetParameterDescriptorComparands)
-                .ToList();
-
-            return ModelDescriptor.ConstructorDescriptors
-                .FirstOrDefault(matchCtorDesc =>
-                {
-                    if (matchCtorDesc.Parent.ModelType != constructorInfo.DeclaringType)
-                        return false;
-                    return matchCtorDesc.ParameterDescriptors
-                        .Select(GetParameterDescriptorComparands)
-                        .SequenceEqual(cmpParamDescs);
-                });
-
-            // Name matching is not necessary for overload descisions.
-            static (Type paramType, bool allowNull, bool hasDefaultValue)
-                GetParameterDescriptorComparands(ParameterDescriptor desc) =>
-                (desc.ValueType, desc.AllowsNull, desc.HasDefaultValue);
-        }
-
-        protected IValueDescriptor FindModelPropertyDescriptor(
-            Type propertyType, string propertyName)
+        // Consider deprecating in favor or BindingConfiguration/BindingContext attach validatation. Then make internal.
+        // Or at least rename to "ConfigureBinding" or similar
+        public void BindConstructorArgumentFromValue(ParameterInfo parameter, IValueDescriptor valueDescriptor)
         {
-            return ModelDescriptor.PropertyDescriptors
-                .FirstOrDefault(desc =>
-                    desc.ValueType == propertyType &&
-                    string.Equals(desc.ValueName, propertyName, StringComparison.Ordinal)
-                    );
-        }
-
-        public void BindConstructorArgumentFromValue(ParameterInfo parameter,
-            IValueDescriptor valueDescriptor)
-        {
-            if (!(parameter.Member is ConstructorInfo constructor))
-                throw new ArgumentException(paramName: nameof(parameter),
-                    message: "Parameter must be declared on a constructor.");
-
+            var constructor = FindConstructorOrThrow(parameter, "Parameter must be declared on a constructor.");
```

Comment:
```suggestion
            var constructor = parameter.Member as ConstructorInfo ?? throw new ArgumentException(paramName: nameof(parameter),
                      message: "Parameter must be declared on a constructor.");
```

This also allows the removal of `FindConstructorOrThrow`